platform/features/layout/src/FixedController.js

```diff
@@ -311,26 +288,225 @@ define(
+         * @param {object[]} objects
+         * @returns {object[]} the provided objects for chaining.
+         */
+        FixedController.prototype.fetchHistoricalData = function (objects) {
+            var bounds = this.openmct.conductor.bounds();
+            var self = this;
+
+            objects.forEach(function (object) {
+                self.openmct.telemetry.request(object, {start: bounds.start, end: bounds.end, size: 1})
+                    .then(self.receiveTelemetry.bind(self, object));
+            });
+            return objects;
+        };
+
+
+        /**
+         * Print a value to the onscreen element associated with a given telemetry object.
+         * @private
+         * @param {object} telemetryObject The telemetry object associated with the value
+         * @param {string | number} value The value to print to screen
+         * @param {string} [cssClass] an optional CSS class to apply to the onscreen element.
+         */
+        FixedController.prototype.setDisplayedValue = function (telemetryObject, value, cssClass) {
+            var id = objectUtils.makeKeyString(telemetryObject.identifier);
+            var self = this;
+
+            (self.elementProxiesById[id] || []).forEach(function (element) {
+                self.names[id] = telemetryObject.name;
+                self.values[id] = value;
+                element.name = self.names[id];
+                element.value = self.values[id];
+                element.cssClass = cssClass;
+            });
+        };
+
+        /**
+         * Process a telemetry datum received for an individual object
+         * @private
+         * @param {object} telemetryObject the domain object associated with the telemetry datum
+         * @param {object} datum a telemetry datum
+         */
+        FixedController.prototype.receiveTelemetry = function (telemetryObject, datum) {
+            this.getCollectionForObject(telemetryObject).add(datum);
+        };
+
+        FixedController.prototype.getTelemetry = function (domainObject) {
+            var newObject = domainObject.useCapability('adapter');
+            var self = this;
+
+            if (this.subscriptions.length > 0) {
+                this.unsubscribe();
+            }
+
+            function filterForTelemetryObjects(objects) {
+                return objects.filter(function (object) {
+                    return self.openmct.telemetry.canProvideTelemetry(object);
+                });
+            }
+
+            return this.openmct.composition.get(newObject).load()
+                .then(filterForTelemetryObjects)
+                .then(this.fetchHistoricalData)
+                .then(this.subscribeToObjects)
+                .then(function (objects) {
+                    self.telemetryObjects = objects;
+                    objects.forEach(function (object) {
+                        // Initialize values
+                        self.setDisplayedValue(object, "");
```

Comment:
Ah! yes, very good point. Race condition aside, it makes the intent (initialization) clearer if we move it before subscription etc.